src/lib/TopLevel.hs

```diff
@@ -569,15 +570,15 @@ evalBlock typed = do
   simplifiedBlock <- checkPass SimpPass $ simplifyTopBlock synthed
   evalRequiredSpecializations simplifiedBlock
   SimplifiedBlock simp recon <- return simplifiedBlock
-  opt <- whenOpt simp $ checkPass OptPass . optimize
+  simp' <- hoistIxDicts simp
+  opt <- whenOpt simp' $ checkPass OptPass . optimize
   result <- case opt of
     AtomicBlock result -> return result
     _ -> do
-      explicitIx <- emitIx opt
-      instIx <- simplifyIx explicitIx
-      lowered <- checkPass LowerPass $ lowerFullySequential instIx
+      lowered <- checkPass LowerPass $ lowerFullySequential opt
+      evalRequiredSpecializations lowered
```

Comment:
I'm not loving the fact that this function is called from so many places. And even _twice_ in this function! I can't prove to myself that this set of calls is necessary or sufficient. What's the right way to think about this?

Reply:
Follow-up: yes, this particular call to `evalRequiredSpecializations` isn't needed.